aten/src/ATen/native/cpu/SortingKernel.cpp

```diff
@@ -167,6 +179,107 @@ static inline void sort_kernel_impl(const value_accessor_t& value_accessor,
   }
 }
 
+#if defined(XSS_COMPILE_TIME_SUPPORTED)
+
+#define AT_DISPATCH_CASE_XSS_TYPES(...)          \
+  AT_DISPATCH_CASE(at::ScalarType::Long, __VA_ARGS__) \
+  AT_DISPATCH_CASE(at::ScalarType::Int, __VA_ARGS__) \
+  AT_DISPATCH_CASE(at::ScalarType::Double, __VA_ARGS__) \
+  AT_DISPATCH_CASE(at::ScalarType::Float, __VA_ARGS__)
+
+#define AT_DISPATCH_XSS_TYPES(TYPE, NAME, ...) \
+  AT_DISPATCH_SWITCH(TYPE, NAME, AT_DISPATCH_CASE_XSS_TYPES(__VA_ARGS__))
+
+static bool can_use_xss_sort(const TensorBase& values, const TensorBase& indices, int64_t dim, const bool stable) {
+  // xss_sort is not a stable sort
+  if (stable) return false;
+
+  auto type = values.scalar_type();
+  if (! (type == ScalarType::Long || type == ScalarType::Int || type == ScalarType::Double || type == ScalarType::Float)) return false;
+
+  return true;
+}
+
+static void xss_sort_kernel(
+    const TensorBase& values,
+    const TensorBase& indices,
+    int64_t dim,
+    bool descending) {
+  auto iter = TensorIteratorConfig()
+    .check_all_same_dtype(false)
+    .resize_outputs(false)
+    .declare_static_shape(values.sizes(), /*squash_dims=*/dim)
+    .add_output(values)
+    .add_output(indices)
+    .build();
+
+  using index_t = int64_t;
+
+  AT_DISPATCH_XSS_TYPES(values.scalar_type(), "xss_sort_kernel", [&] {
+
+    auto values_dim_stride = values.stride(dim);
+    auto indices_dim_stride = indices.stride(dim);
+    auto dim_size = values.size(dim);
+
+    auto loop = [&](char** data, const int64_t* strides, int64_t n) {
+      auto* values_data_bytes = data[0];
+      auto* indices_data_bytes = data[1];
+
+      if(values_data_bytes==nullptr || indices_data_bytes==nullptr){
+        return;
+      }
+
+      if (values_dim_stride == 1 && indices_dim_stride == 1){
+        for (const auto i C10_UNUSED : c10::irange(n)) {
+          x86simdsortStatic::keyvalue_qsort<scalar_t, index_t>(
+              reinterpret_cast<scalar_t*>(values_data_bytes),
+              reinterpret_cast<index_t*>(indices_data_bytes),
+              dim_size,
+              true,
+              descending);
+
+          values_data_bytes += strides[0];
+          indices_data_bytes += strides[1];
+        }
+      }else{
+        std::vector<scalar_t> tmp_values(dim_size);
+        std::vector<index_t> tmp_indices(dim_size);
```

Comment:
`std::vector` zero-fills, would be better to use `c10::SmallBuffer`

Reply:
Done!